### PR TITLE
Handle http/https endpoints

### PR DESCRIPTION
**Pull request by prankulmahajan**
The PR does the following:

1. Decode encoded api_key in case of satellite environment.
2. Handle response if URL is of the form "http" instead of "https"
3. Add vpc-id label in case of satellite cluster

### CODE REVIEW

**Review by ambiknai — APPROVED** [review]

**Review by sameshai — APPROVED** [review]
LGTM

File: pkg/utils/constants.go [hunk 1]
```diff
@@ -61,6 +61,9 @@ const (
 	// NodeWorkerIDLabel  worker ID label attached to node
 	NodeWorkerIDLabel = "ibm-cloud.kubernetes.io/worker-id"
 
+	// NodeInstanceIDLabel VPC ID label attached to satellite host
+	NodeInstanceIDLabel = "ibm-cloud.kubernetes.io/vpc-instance-id"
```
Comment (arahamad) [inline pkg/utils/constants.go:65]: please add this label in CA as well

**Review by arahamad — APPROVED** [review]
lgtm, 

We are adding new labels in the worker label i.e `“ibm-cloud.kubernetes.io/vpc-instance-id”` which we need to consider in the CA, so new patch can add this label in CA

File: pkg/utils/constants.go (hunk 1 above)
Comment (prankulmahajan) [reply pkg/utils/constants.go:65]: Noted !

**Pull request merged**